src/api/c/moments.cpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************
+ * Copyright (c) 2016, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/moments.h>
+#include <af/index.h>
+#include <af/data.h>
+
+#include <ArrayInfo.hpp>
+#include <graphics_common.hpp>
+#include <err_common.hpp>
+#include <backend.hpp>
+#include <moments.hpp>
+#include <handle.hpp>
+#include <reorder.hpp>
+#include <tile.hpp>
+#include <join.hpp>
+#include <cast.hpp>
+#include <arith.hpp>
+
+#include <iostream>
+#include <limits>
+
+using af::dim4;
+using namespace detail;
+
+template<typename T>
+static inline void moments(af_array *out, const af_array in, af_moment_type moment)
+{
+    Array<float> temp = moments<T>(getArray<T>(in), moment);
+    af_array tarr = getHandle<float>(temp);
+    Array<T> output = castArray<T>(tarr);
+
+    *out = getHandle<T>(output);
+}
+
+af_err af_moments(af_array *out, const af_array in, const af_moment_type moment)
+{
+    try {
+        const ArrayInfo in_info = getInfo(in);
+        af_dtype type = in_info.getType();
```

Comment:
Doesn't this function need more checks like the number of dimensions (2D/3D etc)?